middlewares/auth.middleware.js

```diff
@@ -0,0 +1,65 @@
+import jwt from 'jsonwebtoken';
+import {
+  UnAuthorizedError,
+  NotFoundError,
+  ForbiddenError,
+} from '../helpers/errors.js';
+import { roles } from '../constants/index.js';
+import User from '../models/user/index.js';
+
+const SECRET_KEY = process.env.SECRET_KEY;
+
+/**
+ * Middleware for role-based JWT token verification.
+ *
+ * @param {Object} options - Middleware options.
+ * @param {string} options.role - Required user role ('user', 'chef', 'courier', 'admin').
+ *
+ * @example
+ * // Protect a route requiring 'chef' role
+ * router.get('/protectedRoute', verifyToken({ role: 'chef' }), async (req, res) => {
+ *   // Your protected route logic here
+ * });
+ *
+ * @returns {string} - Required roleId ('userId', 'chefId', etc.) if the token is valid.
+ */
+
+const verifyToken = ({ role }) => {
+  const getTokenFromHeaders = (req) => req.headers.authorization?.split(' ')[1];
+
+  const getRoleId = (userRoles) =>
+    userRoles.find((userRole) => userRole.name === role)?.id;
```

Comment:
`name === id`?
there should be an error either in params naming or in logic, doesn't `id` identify an `_id` of other collection?